microcosm_pubsub/registry.py

```diff
@@ -109,11 +109,19 @@ def compute_bound_handlers(self, bindings):
             type(bound_component)
             for bound_component in bound_components
         ]
-        return {
-            media_type: handler
-            for media_type, handler in self.iter_handlers()
-            if handler in bound_components or handler in bound_component_types
-        }
+
+        bound_handlers = dict()
+        for media_type, handler in self.iter_handlers():
+            if handler in bound_components or handler in bound_component_types:
+                if bound_handlers.get(media_type):
+                    raise AlreadyRegisteredError(
+                        "Handler {} already registered for media type: {}".format(
+                            bound_handlers.get(media_type).__name__,
+                            media_type,
+                        )
+                    )
+                bound_handlers[media_type] = handler
```

Comment:
There was a comment last time about making this a stream and still using the dict comprehension. I couldn't think of a nice way to do that without essentially re-doing the already-exists computation that I'm doing here.